service/src/main/java/org/cbioportal/service/impl/StudyViewServiceImpl.java

```diff
@@ -61,4 +71,101 @@
             .collect(Collectors.toList());
     }
 
+    @Override
+    public List<AlterationCountByGene> getMutationAlterationCountByGenes(List<String> studyIds, List<String> sampleIds)
+        throws StudyNotFoundException {
+        if (CollectionUtils.isEmpty(studyIds) || CollectionUtils.isEmpty(sampleIds)) {
+            return new ArrayList<>();
+        }
+        List<MolecularProfileCaseIdentifier> caseIdentifiers =
+            molecularProfileService.getFirstMutationProfileCaseIdentifiers(studyIds, sampleIds);
+        List<AlterationCountByGene> alterationCountByGenes = alterationCountService.getSampleMutationCounts(
+            caseIdentifiers,
+            Select.all(),
+            true,
+            false,
+            Select.all()).getFirst();
+        annotateDataWithQValue(studyIds, alterationCountByGenes);
+        return alterationCountByGenes;
+    }
+
+    @Override
+    public List<AlterationCountByGene> getStructuralVariantAlterationCountByGenes(List<String> studyIds, List<String> sampleIds)
+        throws StudyNotFoundException {
+        if (CollectionUtils.isEmpty(studyIds) || CollectionUtils.isEmpty(sampleIds)) {
+            return new ArrayList<>();
+        }
+        List<MolecularProfileCaseIdentifier> caseIdentifiers =
+            molecularProfileService.getFirstStructuralVariantProfileCaseIdentifiers(studyIds, sampleIds);
+        List<AlterationCountByGene> alterationCountByGenes = alterationCountService.getSampleStructuralVariantCounts(
+            caseIdentifiers,
+            Select.all(),
+            true,
+            false).getFirst();
+        annotateDataWithQValue(studyIds, alterationCountByGenes);
+        return alterationCountByGenes;
+    }
+
+    private void annotateDataWithQValue(List<String> studyIds, List<AlterationCountByGene> alterationCountByGenes)
+        throws StudyNotFoundException {
+        Set<String> distinctStudyIds = new HashSet<>(studyIds);
+        if (!alterationCountByGenes.isEmpty() && distinctStudyIds.size() == 1) {
```

Comment:
So if you're querying for more than one study, you don't get q values? What's the reasoning here?

Reply:
Yes. `alterationCountByGenes` -> these are aggregated counts(count per gene may be coming from multiple studies).  there might be multiple q-values for multi-study query